main.sh

```diff
@@ -1,126 +1,528 @@
+  phpcs_standards_to_ignore='PHPCSUtils'
 else
-  phpcs_file_path="--phpcs-path='/home/rtbot/vip-go-ci-tools/phpcs/bin/phpcs'"
-fi
-
-[[ -z "$PHPCS_SNIFFS_EXCLUDE" ]] && phpcs_sniffs_exclude='' || phpcs_sniffs_exclude="--phpcs-sniffs-exclude='$PHPCS_SNIFFS_EXCLUDE'"
-
-[[ -z "$SKIP_FOLDERS" ]] && skip_folders_option='' || skip_folders_option="--skip-folders='$SKIP_FOLDERS'"
-
-/usr/games/cowsay "Running with the flag $phpcs_standard"
-
-php_lint_option='--lint=true'
-if [[ "$(echo "$PHP_LINT" | tr '[:upper:]' '[:lower:]')" = 'false' ]]; then
-  php_lint_option='--lint=false'
-fi
-
-echo "Running the following command"
-echo "/home/rtbot/vip-go-ci-tools/vip-go-ci/vip-go-ci.php \
-  --phpcs-skip-folders-in-repo-options-file=true \
-  --lint-skip-folders-in-repo-options-file=true \
-  --repo-options=true \
-  --phpcs=true \
-  --repo-owner=$GITHUB_REPO_OWNER \
-  --repo-name=$GITHUB_REPO_NAME \
-  --commit=$COMMIT_ID \
-  --token=\$GH_BOT_TOKEN \
-  --local-git-repo=$RTBOT_WORKSPACE \
-  $phpcs_file_path \
-  $phpcs_standard \
-  $phpcs_sniffs_exclude \
-  $skip_folders_option \
-  $php_lint_option \
-  --informational-url='https://github.com/rtCamp/action-phpcs-code-review/'"
-
-gosu rtbot bash -c \
-  "/home/rtbot/vip-go-ci-tools/vip-go-ci/vip-go-ci.php \
-  --phpcs-skip-folders-in-repo-options-file=true \
-  --lint-skip-folders-in-repo-options-file=true \
-  --repo-options=true \
-  --phpcs=true \
-  --repo-owner=$GITHUB_REPO_OWNER \
-  --repo-name=$GITHUB_REPO_NAME \
-  --commit=$COMMIT_ID \
-  --token=$GH_BOT_TOKEN \
-  --local-git-repo=$RTBOT_WORKSPACE \
-  $phpcs_file_path \
-  $phpcs_standard \
-  $phpcs_sniffs_exclude \
-  $skip_folders_option \
-  $php_lint_option \
-  --informational-url='https://github.com/rtCamp/action-phpcs-code-review/'"
+  phpcs_standards_to_ignore="$PHPCS_STANDARDS_TO_IGNORE"
+fi
+
+CMD+=( "--phpcs-standards-to-ignore=$phpcs_standards_to_ignore" )
+
+#######################################
+# Set the --phpcs-skip-scanning-via-labels-allowed
+# Default: true
+# Options: BOOLEAN
+#######################################
+phpcs_skip_scanning_via_labels_allowed='true'
+if [[ "$PHPCS_SKIP_SCANNING_VIA_LABELS_ALLOWED" == "false" ]]; then
+  phpcs_skip_scanning_via_labels_allowed='false'
+fi
+
+CMD+=( "--phpcs-skip-scanning-via-labels-allowed=$phpcs_skip_scanning_via_labels_allowed" )
+
+#######################################
+# Set the --phpcs-skip-folders
+# Default: vendor,node_modules
+# Options: STRING (Comma separated list of folders to skip)
+#######################################
+if [[ -z "$PHPCS_SKIP_FOLDERS" ]]; then
+  phpcs_skip_folders='vendor,node_modules'
+else
+  phpcs_skip_folders="$PHPCS_SKIP_FOLDERS"
+fi
+
+# Keep SKIP_FOLDERS for backward compatibility
+if [[ -n "$SKIP_FOLDERS" ]]; then
+  if [[ -n "$phpcs_skip_folders" ]]; then
+    phpcs_skip_folders="$phpcs_skip_folders,"
+  fi
+
+  phpcs_skip_folders="$phpcs_skip_folders$SKIP_FOLDERS"
+fi
+
+CMD+=( "--phpcs-skip-folders=$phpcs_skip_folders" )
```

Comment:
This skips only the immediate folder given, but if there are any subfolders in it, which will be the case for `tests`, `vendor` or other plugin folders, then they will not be skipped and will lead to unnecessary errors. So, bring back the original code for `SKIP_FOLDERS` which removes the entire folder to avoid scan.

Reply:
The behavior of both `--phpcs-skip-folders` and 

https://github.com/rtCamp/action-phpcs-code-review/blob/68967fd44a8f7000473bcbd6b282af9bc89fe0f3/main.sh#L21-L29

is same. Both will ignore the specific path provided. For example, consider below file structure
```
- src
    |- vendor
- vendor
    |- vendor
```
If you pass `vendor` to be ignored, both will miss the `vendor`, `vendor/vendor` directories and not the `src/vendor`.

Reply:
```
- vendor
    |- vendor-1/php-file-with-errors-1.php
    |- vendor-2/php-file-with-errors-2.php
```

In this case, `--phpcs-skip-folders` will scan and give errors in both the PHP files. Whereas the other method will not.

Reply:
This was an implementation error. `--phpcs-skip-folders` is working as expected as per the latest changes.